framework/sysdeps/windows/tmpfile.c

```diff
@@ -40,9 +41,68 @@ static unsigned next_random()
     return result;
 }
 
+static bool is_dir_writable(const wchar_t *dir)
+{
+    if (!dir)
+        return false;
+
+    size_t len = wcslen(dir);
+    if (!len)
+        return false;
+    if (dir[len - 1] == L'\\')
+        return _waccess(dir, W_OK) == 0;
+
+    // append a backslash so we confirm it's a directory
+    wchar_t path[MAX_PATH];
+    wcscpy(path, dir);
+    wcscat(path, L"\\");
+    return _waccess(path, W_OK) == 0;
+}
+
+static const wchar_t *find_alternate_temp_dir()
+{
+    static const wchar_t *dirname = NULL;
+    if (dirname)
+        return dirname;
+
+    dirname = _wgetenv(L"TEMP");
+    if (is_dir_writable(dirname))
+        return dirname;
+
+    dirname = _wgetenv(L"TMP");
+    if (is_dir_writable(dirname))
+        return dirname;
+
```

Comment:
Shouldn't we set `dirname` back to `NULL` if `is_dir_writable(L"TMP")` check fails?